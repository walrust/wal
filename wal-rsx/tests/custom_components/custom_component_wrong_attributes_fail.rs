use wal_core::{
    component::{behavior::Behavior, Component},
    virtual_dom::VNode,
};
use wal_rsx::rsx;

include!("../utils/custom_components/custom_component_props_i32.rs");
include!("../utils/non_display_struct.rs");

fn main() {
    rsx! { <CustomComponentPropsI32 attr = 5 /> };
    rsx! { <CustomComponentPropsI32 props = 1 attr = 5 /> };
    rsx! { <CustomComponentPropsI32 key = "key" attr = 5 /> };
    rsx! { <CustomComponentPropsI32 props = 1 attr = 5 key="key" /> };
    rsx! { <CustomComponentPropsI32 props = 0 props = 1 /> };
    rsx! { <CustomComponentPropsI32 props = 0 key = "key" props = 1 /> };
    rsx! { <CustomComponentPropsI32 key = "key1" key = "key2" /> };
    rsx! { <CustomComponentPropsI32 key = "key1" props = 0 key = "key2" /> };
    rsx! { <CustomComponentPropsI32 props /> };
    rsx! { <CustomComponentPropsI32 props = /> };
    rsx! { <CustomComponentPropsI32 props = key = "key" /> };
    rsx! { <CustomComponentPropsI32 key /> };
    rsx! { <CustomComponentPropsI32 key = /> };
    rsx! { <CustomComponentPropsI32 key = props = 1 /> };
    rsx! { <CustomComponentPropsI32 key = {NonDisplayStruct} /> };
    rsx! { <CustomComponentPropsI32 props = "value" /> };
    rsx! { <CustomComponentPropsI32 props = {"value"} /> };
    rsx! { <CustomComponentPropsI32 props = NonDisplayStruct {} /> };
}
