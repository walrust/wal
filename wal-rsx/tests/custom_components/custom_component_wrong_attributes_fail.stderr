error: Unsupported attribute `attr`. Custom components supports only `props` and `key` attributes
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:11:37
   |
11 |     rsx! { <CustomComponentPropsI32 attr = 5 /> };
   |                                     ^^^^

error: Unsupported attribute `attr`. Custom components supports only `props` and `key` attributes
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:12:47
   |
12 |     rsx! { <CustomComponentPropsI32 props = 1 attr = 5 /> };
   |                                               ^^^^

error: Unsupported attribute `attr`. Custom components supports only `props` and `key` attributes
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:13:49
   |
13 |     rsx! { <CustomComponentPropsI32 key = "key" attr = 5 /> };
   |                                                 ^^^^

error: Unsupported attribute `attr`. Custom components supports only `props` and `key` attributes
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:14:47
   |
14 |     rsx! { <CustomComponentPropsI32 props = 1 attr = 5 key="key" /> };
   |                                               ^^^^

error: Duplicate attribute `props`
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:15:47
   |
15 |     rsx! { <CustomComponentPropsI32 props = 0 props = 1 /> };
   |                                               ^^^^^

error: Duplicate attribute `props`
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:16:59
   |
16 |     rsx! { <CustomComponentPropsI32 props = 0 key = "key" props = 1 /> };
   |                                                           ^^^^^

error: Duplicate attribute `key`
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:17:50
   |
17 |     rsx! { <CustomComponentPropsI32 key = "key1" key = "key2" /> };
   |                                                  ^^^

error: Duplicate attribute `key`
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:18:60
   |
18 |     rsx! { <CustomComponentPropsI32 key = "key1" props = 0 key = "key2" /> };
   |                                                            ^^^

error: expected `=`
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:19:43
   |
19 |     rsx! { <CustomComponentPropsI32 props /> };
   |                                           ^

error: Expected a literal, a struct literal expression or an expression block
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:20:45
   |
20 |     rsx! { <CustomComponentPropsI32 props = /> };
   |                                             ^

error: Expected a literal, a struct literal expression or an expression block
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:21:45
   |
21 |     rsx! { <CustomComponentPropsI32 props = key = "key" /> };
   |                                             ^^^

error: expected `=`
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:22:41
   |
22 |     rsx! { <CustomComponentPropsI32 key /> };
   |                                         ^

error: Expected a literal or an expression block
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:23:43
   |
23 |     rsx! { <CustomComponentPropsI32 key = /> };
   |                                           ^

error: Expected a literal or an expression block
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:24:43
   |
24 |     rsx! { <CustomComponentPropsI32 key = props = 1 /> };
   |                                           ^^^^^

error[E0599]: `NonDisplayStruct` doesn't implement `std::fmt::Display`
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:25:43
   |
25 |     rsx! { <CustomComponentPropsI32 key = {NonDisplayStruct} /> };
   |                                           ^^^^^^^^^^^^^^^^^^ `NonDisplayStruct` cannot be formatted with the default formatter
   |
  ::: tests/custom_components/../utils/non_display_struct.rs
   |
   | pub struct NonDisplayStruct;
   | ---------------------------
   | |
   | method `to_string` not found for this struct
   | doesn't satisfy `NonDisplayStruct: ToString`
   | doesn't satisfy `NonDisplayStruct: std::fmt::Display`
   |
   = note: the following trait bounds were not satisfied:
           `NonDisplayStruct: std::fmt::Display`
           which is required by `NonDisplayStruct: ToString`
note: the trait `std::fmt::Display` must be implemented
  --> $RUST/core/src/fmt/mod.rs
   |
   | pub trait Display {
   | ^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:26:45
   |
26 |     rsx! { <CustomComponentPropsI32 props = "value" /> };
   |            -                                ^^^^^^^ expected `i32`, found `&str`
   |            |
   |            arguments to this function are incorrect
   |
note: associated function defined here
  --> $WORKSPACE/wal/src/virtual_dom/vcomponent.rs
   |
   |     pub fn new<C>(props: C::Properties, key: Option<String>) -> VComponent
   |            ^^^

error[E0308]: mismatched types
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:27:46
   |
27 |     rsx! { <CustomComponentPropsI32 props = {"value"} /> };
   |                                              ^^^^^^^ expected `i32`, found `&str`

error[E0308]: mismatched types
  --> tests/custom_components/custom_component_wrong_attributes_fail.rs:28:45
   |
28 |     rsx! { <CustomComponentPropsI32 props = NonDisplayStruct {} /> };
   |            -                                ^^^^^^^^^^^^^^^^^^^ expected `i32`, found `NonDisplayStruct`
   |            |
   |            arguments to this function are incorrect
   |
note: associated function defined here
  --> $WORKSPACE/wal/src/virtual_dom/vcomponent.rs
   |
   |     pub fn new<C>(props: C::Properties, key: Option<String>) -> VComponent
   |            ^^^
